src/Maker/MakeMiddleware.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ *
+ * @internal
+ */
+final class MakeMiddleware extends AbstractMaker
+{
+    public static function getCommandName(): string
+    {
+        return 'make:middleware';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->setDescription('Creates a new messenger middleware')
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the middleware class (e.g. <fg=yellow>CustomMiddleware</>)')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeMessage.txt'));
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $middlewareClassNameDetails = $generator->createClassNameDetails(
+            $input->getArgument('name'),
+            'Middleware\\',
+            'Middleware'
+        );
+
+        $generator->generateClass(
+            $middlewareClassNameDetails->getFullName(),
+            'middleware/Middleware.tpl.php'
+        );
+
+        $generator->writeChanges();
+
+        $this->writeSuccessMessage($io);
+
+        $io->text([
+            'Next: Open your new middleware class and add the code you need.',
+            'Find the documentation at <fg=yellow>https://symfony.com/doc/current/messenger.html</>',
```

Comment:
I guess we could link straight to: https://symfony.com/doc/current/messenger.html#middleware

Reply:
And there is a second step we need to mention: you need to add this middleware to your `messenger.yaml` file. But I don't think we should attempt to add this automatically in this command - there are just too many use-cases. But we should mention (in this message) that you need to add it.